src/EditorFeatures/Core/Intents/IIntentProcessor.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Features.Intents;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.VisualStudio.Text;
+
+namespace Microsoft.CodeAnalysis.Editor.Intents
+{
+    internal interface IIntentProcessor
```

Comment:
@Intellicode this file contains the rough API shape that we've gotten.  We have a few questions that came up during the implementation.  I'm adding github comments in this file for each item.

1.  My plan was to put wrappers in pythia external access, are you able to consume them from there?

Reply:
The pythia is running out of proc in an other process so I'm not sure if we are able to. I'll verify with vivian to be sure.

Reply:
@pgroene these look like the dlls that have IVTs - https://github.com/dotnet/roslyn/blob/6da1274c9d24c2f90a48290394a951b23617f2a3/src/Features/Core/Portable/Microsoft.CodeAnalysis.Features.csproj#L99
"Microsoft.VisualStudio.IntelliCode.CSharp"
"Microsoft.VisualStudio.IntelliCode.CSharp.Extraction"

Reply:
Can we add the Microsoft.VisualStudio.IntelliCode our VSIX package?

We do not have direct access to those other dlls, and it seems that taking the direct reference makes sense here

Reply:
@pgroene Yeah I think it should be possible to add another dll there.  Once we agree on the shape I'll update this PR to include that dll.

Reply:
Added restricted ivt for the Microsoft.VisualStudio.IntelliCode dll.  This should allow access to everything in the MS.CA.ExternalAccess.IntelliCode.Api namespace